pandas/tseries/tools.py

```diff
@@ -283,10 +303,25 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     NaT
 
     """
-    return _to_datetime(arg, errors=errors, dayfirst=dayfirst,
-                        yearfirst=yearfirst,
-                        utc=utc, box=box, format=format, exact=exact,
-                        unit=unit, infer_datetime_format=infer_datetime_format)
+    # variable to set offset as per origin parameter
+    offset = None
+
+    if origin == 'julian':
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        arg = arg - tslib.Timestamp(0).to_julian_date()
```

Comment:
subtracting the arg may not make sense here, in fact shouldn't it be converted to a julian date?

need some more testing with ``origin='julian'``

```
In [30]: pd.to_datetime(pd.to_datetime([1,2],unit='D').to_julian_date()-pd.Timestamp(0).to_julian_date(),unit='D')
Out[30]: DatetimeIndex(['1970-01-02', '1970-01-03'], dtype='datetime64[ns]', freq=None)
```